s3/pkg/meta/db/drivers/tidb/bucket.go

```diff
@@ -267,32 +293,28 @@ func (t *TidbClient) ListObjects(ctx context.Context, bucketName, marker, verIdM
 				objectNum[name] = 0
 			}
 			objectNum[name] += 1
-			marker = name
-			//filte row
-			//filte by prefix
-			hasPrefix := strings.HasPrefix(name, prefix)
-			if !hasPrefix {
-				continue
-			}
-			//filte by objectname
+			filter[common.KMarker] = name
+
 			if _, ok := objectMap[name]; !ok {
 				objectMap[name] = struct{}{}
 			} else {
 				continue
 			}
-			//filte by deletemarker
-			if deletemarker {
-				continue
-			}
+
+			// TODO: filter by deletemarker if versioning enabled
+
 			if name == omarker {
+				log.Infof("%s euqals to omarker, continue\n", name)
 				continue
 			}
-			//filte by delemiter
+
+			// group by delimiter
 			if len(delimiter) != 0 {
 				subStr := strings.TrimPrefix(name, prefix)
 				n := strings.Index(subStr, delimiter)
 				if n != -1 {
 					prefixKey := prefix + string([]byte(subStr)[0:(n+1)])
+					filter[common.KMarker] = prefixKey[0:(len(prefixKey)-1)] + string(delimiter[len(delimiter)-1]+1)
```

Comment:
please add comment to explain what this code is supposed to do, thanks

Reply:
Done.